components/user_settings/advanced/user_settings_advanced.jsx

```diff
@@ -57,7 +57,12 @@ export default class AdvancedSettingsDisplay extends React.PureComponent {
             join_leave: this.props.joinLeave,
         };
 
-        const preReleaseFeaturesKeys = Object.keys(PreReleaseFeatures);
+        const PreReleaseFeaturesLocal = JSON.parse(JSON.stringify(PreReleaseFeatures));
```

Comment:
Since, `PreReleaseFeatures` is a global variable and it looks like this `delete` key effect should also be global. So, I think it would be better to actually modify the original `PreReleaseFeatures` in this case, otherwise it could create a "potential" bug like in this line `const feature = PreReleaseFeatures[key];`. It is still accessing the original variable.